### PR TITLE
Allow access to Active Directory from the registration vlan

### PR DESCRIPTION
**Pull request by fdurand**
# Description

Allow windows devices on the registration vlan to be able to reach Active Directory
# Impacts

No
# Delete branch after merge

YES
# NEWS file entries
## New Features
- Windows devices on the registration vlan are now able to reach there Active Directory (Must be configure in PacketFence's domain section)

### CODE REVIEW

**Comment by julsemaan** [comment]
The amount of people we'll make happy with this...
Very nice

File: sbin/pfdns [hunk 1]
```diff
@@ -137,6 +138,12 @@ foreach my $network ( keys %ConfigNetworks ) {
     }
 }
 
+# Instantiate dns_hash
+my %dns_hash;
```
Comment (julsemaan) [inline sbin/pfdns]: I suggest you create a pfconfig resource for that and have it expired by config::Domain

This way, we won't need to restart pfdns when we activate the passthrough for this domain

File: html/pfappserver/lib/pfappserver/Form/Config/Domain.pm [hunk 2]
```diff
@@ -96,6 +96,14 @@ has_field 'dns_name' =>
              help => 'The DNS name (FQDN) of the domain.' },
   );
 
+has_field 'registration' =>
+  (
+   type => 'Checkbox',
+   label => 'Allow on registration',
+   tags => { after_element => \&help,
+             help => 'If this option is enabled, the device will be able to reach the Active Directory from the registration.' },
```
Comment (julsemaan) [inline html/pfappserver/lib/pfappserver/Form/Config/Domain.pm]: *registration VLAN

As if they are isolated by web auth it won't work

File: html/pfappserver/root/config/domain/view.tt [hunk 3]
```diff
@@ -17,6 +17,7 @@
         [% form.field('dns_server').render | none %]
         [% form.field('bind_dn').render | none %]
         [% form.field('bind_pass').render | none %]
+        [% form.field('registration').render %]
```
Comment (julsemaan) [inline html/pfappserver/root/config/domain/view.tt]: | none

**Comment by lzammit** [comment]
Looks good to me, need to be tested.

**Comment by jrouzierinverse** [comment]
Then test it

**Comment by jrouzierinverse** [comment]
then test it

James Rouzier
jrouzier@inverse.ca :: +1.514.447.4918 (x115)  ::  http://www.inverse.ca
Inverse inc. :: Leaders behind SOGo (http://www.sogo.nu) and PacketFence (http://www.packetfence.org)

**Comment by dwlfrth** [comment]
@lzammit : I think @jrouzierinverse wants you to test it ;)

**Comment by fdurand** [comment]
Not sure LDAP SSL TCP 636 is needed, we just want to be able to contact the AD to open a session.
I tested it without this port and it worked.

**Comment by louismunro** [comment]
PR reviews are not expected to test the code.
It's a code review only. Testing is a separate step and should be done both by the developer and the QA person.

**Pull request merged**